esmvalcore/cmor/_fixes/icon/_base_fixes.py

```diff
@@ -271,18 +274,57 @@ def _tmp_local_file(local_file: Path) -> Path:
         with NamedTemporaryFile(prefix=f"{local_file}.") as file:
             return Path(file.name)
 
-    def _get_grid_from_cube_attr(self, cube: iris.cube.Cube) -> iris.cube.Cube:
+    def _get_grid_from_cube_attr(self, cube: Cube) -> Cube:
         """Get horizontal grid from `grid_file_uri` attribute of cube."""
         (grid_url, grid_name) = self._get_grid_url(cube)
 
         # If already loaded, return the horizontal grid
         if grid_name in self._horizontal_grids:
             return self._horizontal_grids[grid_name]
 
-        # Check if grid file has recently been downloaded and load it if
-        # possible
+        # First, check if the grid file is available in the ICON rootpath
+        grid = self._get_grid_from_rootpath(grid_name)
+
+        # Second, if that didn't work, try to download grid (or use cached
+        # version of it if possible)
+        if grid is None:
+            grid = self._get_downloaded_grid(grid_url, grid_name)
+
+        # Cache grid for later use
+        self._horizontal_grids[grid_name] = grid
+
+        return grid
+
+    def _get_grid_from_rootpath(self, grid_name: str) -> CubeList | None:
+        """Try to get grid from the ICON rootpath."""
+        rootpaths = _get_rootpath('ICON')
+        dirname_template = _select_drs('input_dir', 'ICON')
+        dirname_globs = _replace_tags(dirname_template, self.extra_facets)
+        possible_grid_paths = [
+            r / d / grid_name for r in rootpaths for d in dirname_globs
+        ]
+        for grid_path in possible_grid_paths:
+            if grid_path.is_file():
+                logger.debug("Using ICON grid file '%s'", grid_path)
+                return self._load_cubes(grid_path)
```

Comment:
is it always that if `grid_path` is a valid file then it's also a grid file?

Reply:
I should hope so. I've never encountered something that was invalid, and I'm pretty sure we would get some kind of error if the file wasn't valid.

Reply:
I'll take your word for that :grin: